### PR TITLE
Move to Go 1.13

### PR DESCRIPTION
**Pull request by jeffallen**
Change to using Go 1.13, improve documentation about our policy.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-  language: go
-  go: "1.11.x"
 _: &language_go_1_12
   name: "go 1.12"
   language: go
   go: "1.12.x"
+_: &language_go_1_13
+  name: "go 1.12"
+  language: go
+  go: "1.12.x"
```
Comment (nkcr) [inline .travis.yml]: go 1.**13**.x

File: .travis.yml [hunk 2]
```diff
@@ -80,9 +79,9 @@ jobs:
 
     - stage: test
       <<: *stage_test_go
-      <<: *language_go_1_11
-    - <<: *stage_test_go
       <<: *language_go_1_12
```
Comment (ineiti) [inline .travis.yml]: Would this be a good moment to remove the 1_12 testing here, as proposed in #2069 ?

File: .travis.yml (hunk 1 above)
Comment (jeffallen) [reply .travis.yml]: oh god, what an idiot.

File: .travis.yml [hunk 3]
```diff
@@ -79,10 +72,9 @@ jobs:
         - npm run build
 
     - stage: test
-      <<: *stage_test_go
```
Comment (ineiti) [inline .travis.yml]: Travis tries to run some ruby - I wonder if this line should go back in? But I got to admit that @tharvik script-fooed the hell out of .travis.yml and that I never know where a `-` is needed and where not.

Comment (tharvik) [reply .travis.yml]: `<<:` is the merge operator, taking its definition and merging it with the current dict.
```yaml
- stage: test
<<: *stage_test_go
---
- stage: test
  before_script: *get_coding
  script: GO111MODULE=on make test_goveralls
```
```yaml
- stage: test
- <<: *stage_test_go
---
- stage: test
- before_script: *get_coding
  script: GO111MODULE=on make test_goveralls
```

File: .travis.yml (hunk 2 above)
Comment (jeffallen) [reply .travis.yml]: done.

File: .travis.yml [hunk 4]
```diff
@@ -79,10 +75,8 @@ jobs:
         - npm run build
 
     - stage: test
+      <<: *language_go_1_13
```
Comment (tharvik) [inline .travis.yml]: I find it nicer if is `stage` first then `language` (to stay coherent with the others)

Comment (jeffallen) [reply .travis.yml]: But then language gets set to Ruby. Travis is very fragile and mysterious when using the merge operator.

**Pull request merged**